rmgpy/data/kinetics/family.py

```diff
@@ -3068,68 +3095,275 @@ def generateTree(self, obj=None, thermoDatabase=None, T=1000.0):
+                entry.parent.children.append(entry)
+
         return
 
+    def absorbProcess(self,p,conn,name):
+        try:
+            grps = conn.recv()
+            p.terminate()
+        except Exception as e:
+            logging.error('failed to absorb process {}'.format(name))
+            raise e
+        return grps
 
-    def makeBMRulesFromTemplateRxnMap(self, templateRxnMap):
+    def makeBMRulesFromTemplateRxnMap(self,templateRxnMap,nprocs=1,Tref=1000.0,fmax=1.0e5):
 
-        index = max([e.index for e in self.rules.getEntries()] or [0]) + 1
-        
+        ruleKeys = self.rules.entries.keys()
         for entry in self.groups.entries.values():
-            if entry.index == -1 or self.rules.entries[entry.label] != []:
-                continue
-            rxns = templateRxnMap[entry.label]
-            descendants = getAllDescendants(entry)
-            for entry2 in descendants:
-                rxns.extend(templateRxnMap[entry2.label])
-            
-            assert rxns != [], entry.label
-            kinetics = ArrheniusBM().fitToReactions(rxns,family=self)
-            
-            new_entry = Entry(
-                index = index,
-                label = entry.label,
-                item = self.forwardTemplate,
-                data = kinetics,
-                rank = 11,
-                reference=None,
-                shortDesc="BM rule fitted to {0} training reactions at node {1}".format(len(rxns),entry.label),
-                longDesc="BM rule fitted to {0} training reactions at node {1}".format(len(rxns),entry.label),
-            )
-            new_entry.data.comment = "BM rule fitted to {0} training reactions at node {1}".format(len(rxns),entry.label)
+            if entry.label not in ruleKeys:
+                self.rules.entries[entry.label] = []
 
-            self.rules.entries[entry.label].append(new_entry)
-            
-            index += 1
+        index = max([e.index for e in self.rules.getEntries()] or [0]) + 1
+
+        entries = self.groups.entries.values()
+        rxnlists = [(templateRxnMap[entry.label],entry.label) if entry.label in templateRxnMap.keys() else [] for entry in entries]
+        inputs = np.array([(self.forwardRecipe.actions,rxns,Tref,fmax,label,[r.rank for r in rxns]) for rxns,label in rxnlists])
 
+        inds = np.arange(len(inputs))
+        np.random.shuffle(inds) #want to parallelize in random order
+        inds = inds.tolist()
+        revinds = [inds.index(x) for x in np.arange(len(inputs))]
 
+        pool = mp.Pool(nprocs)
 
-    def crossValidate(self, folds=5, templateRxnMap=None, T=1000.0, iters=0, random_state=1):
+        kineticsList = np.array(pool.map(makeRule,inputs[inds]))
+        kineticsList = kineticsList[revinds] #fix order
+
+        for i,kinetics in enumerate(kineticsList):
+            if kinetics is not None:
+                entry = entries[i]
+                std = kinetics.uncertainty.getExpectedLogUncertainty()/0.398
```

Comment:
Where does the 0.398 come from? Could you add a comment?

Reply:
It comes from the fact that 0.398 * std is the expected uncertainty.  Comment added.